xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -117,6 +117,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
```

Comment:
> Please tell me where class CSelectionStreams is accessed from outside of the player thread

see the snippet in my post above: i.e.
CVideoPlayer::GetStreamDetails or
CVideoPlayer::GetAudioStreamInfo
(there is more)

> Am I right to assume that locking in CSelectionStreams is completely unnecessary?

If a user wants to change audio or subtitle track the UI needs to have access to the currently available streams. The UI also accesses this object for getting info to display on the OSD.
Having the GUI poll player data frequently is not ideal. It is much better to provide an extra object for the UI and player updates this object after something has changed.
That's the vision and we don't need to get to final in a single iteration.